src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -75,6 +76,18 @@ private enum ApplicationTypes
             INTERNET_SHORTCUT_APPLICATION = 1,
             WIN32_APPLICATION = 2,
             RUN_COMMAND = 3,
+            FOLDER = 4,
+            GENERIC_FILE = 5,
+        }
+
+        private enum FileTypes
+        {
+            APPLICATION = 0,
+            SHORTCUT = 1,
+            URL = 2,
+            APPREF = 3,
+            FOLDER = 4,
+            GENERIC_FILE = 5,
```

Comment:
I would merge both the enums. I can see that we're setting the applicationtype in a lot of places based on the file type to the same value as the file type. I understand that one is being set in code, depending on the type of application and the other is based on the extension, but if we look closely both are trying to achieve the same thing. I feel we can combine both of them so that we don't set multiple parameters with the same value.

Reply:
I removed the FileTypes enum. The one issue I found is that the only place where Appref, lnk and normal executables were distinguished was in `GetAppNameFromPath`, so for this to work there is some code duplication as we need to check the extension again in this part of the code. Is this fine or do you have any suggestions for avoiding that code duplication? Ideally the `WIN32_APPLICATION` would be split into 3 types: Lnks, Apprefs and normal executables but that would require many more changes.

Reply:
@alekhyareddy28 I have modified it such that now there is only the `ApplicationType` enum which is a combination of FileType and old ApplicationTypes. I also modified the AppType field to ApplicationType instead of the uint values, since tests would otherwise break each time ApplicationTypes was modified. I had to rename some of the fields after making ApplicationTypes public to remove StyleCop/FxCop errors.

Refactored the tests for this and added tests for the new app types.